domain/src/main/java/org/oppia/android/domain/devoptions/ModifyLessonProgressController.kt

```diff
@@ -101,6 +103,83 @@ class ModifyLessonProgressController @Inject constructor(
     return true
   }
 
+  /**
+   * Marks multiple topics completed given a profile ID and list of topic IDs.
+   *
+   * @param profileId the ID corresponding to the profile for which progress needs modified.
+   * @param topicIdList the list of topic IDs for which progress needs modified.
+   * @return a [Boolean] indicating whether the process is completed or not.
```

Comment:
I am curious as to why we need to return the status boolean. And if it fails, how do we know what the issue is?

Reply:
Actually, I was using this boolean just to check if the function call was successful. You can see the use case below https://github.com/oppia/oppia-android/blob/cc53a69992335ff70106799de5f3acd2c35c4bb0/app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/MarkTopicsCompletedFragmentPresenter.kt#L80-L86
But, I think it is unnecessary. I have removed it now. Also, I don't think that there is a possibility for this function to fail as the only reason this might fail will be because of an invalid `topicId`. But since the source of `topicId` is the list of topics fetched using https://github.com/oppia/oppia-android/blob/feb7d812b96c3b384b3ed9bfa38e3e560c058c78/domain/src/main/java/org/oppia/android/domain/devoptions/ModifyLessonProgressController.kt#L39
I don't think there will be a case of invalid `topicId`. WDYT @vinitamurthi? cc: @rt4914 @Sarthak2601.

Reply:
I think even if there is an invalid topicId, this boolean isn't really giving us a lot of information so its better to remove it.